integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextbest/NextBestServiceBus.java

```diff
@@ -0,0 +1,141 @@
+
+import javax.annotation.PostConstruct;
+import javax.xml.bind.*;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+@Component
+public class NextBestServiceBus {
+
+    private static final Logger log = LoggerFactory.getLogger(NextBestServiceBus.class);
+
+    private static final String NEXTBEST_QUEUE_PREFIX = "nextbestqueue";
+
+    private ServiceBusContract service;
+    private IntegrasjonspunktProperties props;
+    private StandardBusinessDocumentFactory sbdf;
+    private MessageSender messageSender;
+    private JAXBContext jaxbContext;
+    private String queuePath;
+
+    @Autowired
+    public NextBestServiceBus(IntegrasjonspunktProperties props,
+                              StandardBusinessDocumentFactory sbdf,
+                              MessageSender messageSender) throws JAXBException {
+
+        this.props = props;
+        this.sbdf = sbdf;
+        this.messageSender = messageSender;
+
+        Configuration config = ServiceBusConfiguration.configureWithSASAuthentication(
+                props.getNextbest().getServiceBus().getNamespace(),
+                "RootManageSharedAccessKey",
+                props.getNextbest().getServiceBus().getSasToken(),
+                ".servicebus.windows.net"
+        );
+        this.service = ServiceBusService.create(config);
+
+        this.jaxbContext = JAXBContext.newInstance(EduDocument.class, Payload.class);
+
+    }
+
+    @PostConstruct
+    private void init() throws ServiceException {
+
+        if (!props.getNextbest().getServiceBus().isEnable()) {
+            return;
+        }
+
+        // Create queue if it does not already exist
+        queuePath = String.format("%s%s%s", NEXTBEST_QUEUE_PREFIX,
+                props.getOrg().getNumber(),
+                props.getNextbest().getServiceBus().getMode());
+        ListQueuesResult queues = service.listQueues();
+        if (!queues.getItems().stream().anyMatch(i -> i.getPath().contains(queuePath))) {
+            log.info("Queue with id {} does not already exist, creating it..", queuePath);
+            QueueInfo qi = new QueueInfo(queuePath);
+            service.createQueue(qi);
+        }
+    }
+
+    public void putMessage(OutgoingConversationResource resource) throws NextBestException {
+
+        BrokeredMessage msg;
+        try {
+            MessageContext context = messageSender.createMessageContext(resource);
+            EduDocument eduDocument = sbdf.create(resource, context);
+
+            Marshaller marshaller = jaxbContext.createMarshaller();
+            ByteArrayOutputStream os = new ByteArrayOutputStream();
+            ObjectFactory of = new ObjectFactory();
+            JAXBElement<EduDocument> sbd = of.createStandardBusinessDocument(eduDocument);
+            marshaller.marshal(sbd, os);
+
+            msg = new BrokeredMessage(os.toByteArray());
+
+            String queuePath = NEXTBEST_QUEUE_PREFIX + resource.getReceiverId();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "queuePath" which hides the field declared at line 44. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AHiddenFieldCheck)